src/pal/inc/pal.h

```diff
@@ -5901,16 +5893,34 @@ typedef VOID (PALAPI *PHARDWARE_EXCEPTION_HANDLER)(PAL_SEHException* ex);
 PALIMPORT
 VOID
 PALAPI
-PAL_SetHardwareExceptionHandler(IN PHARDWARE_EXCEPTION_HANDLER handler);
+PAL_SetHardwareExceptionHandler(
+    IN PHARDWARE_EXCEPTION_HANDLER exceptionHandler);
 
-#endif // __cplusplus
+class PAL_HardwareExceptionHolder
+{
+public:
+    PAL_HardwareExceptionHolder();
+
+    ~PAL_HardwareExceptionHolder();
+
+    static bool PAL_IsEnabled();
```

Comment:
I added the PAL_ prefix because it is a public class and holder like PAL_EnterHolder/PAL_LeaveHolder.

mikem

From: Jan Vorlicek [mailto:notifications@github.com]
Sent: Wednesday, April 22, 2015 1:39 PM
To: dotnet/coreclr
Cc: Mike McLaughlin
Subject: Re: [coreclr] Share hardware exception between coreclr and DAC (#771)


In src/pal/inc/pal.h<https://github.com/dotnet/coreclr/pull/771#discussion_r28912382>:

>

> -#endif // __cplusplus

> +class PAL_HardwareExceptionHolder

> +{

> +public:

> +    PAL_HardwareExceptionHolder();

> +

> +    ~PAL_HardwareExceptionHolder();

> +

> +    static bool PAL_IsEnabled();

I wonder if we really want the PAL_ prefix on this method.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/771/files#r28912382>.